reflection.md

```diff
@@ -15,4 +40,36 @@ Write a few brief paragraphs that discuss:
    you implement it _or_ what features of Scala would prevent you from
    implementing it? (You don't have to write code for this part. You could say
    "I would use literal extension to..." or "Scala's rules for valid
-   identifiers prevent...")
\ No newline at end of file
+   identifiers prevent...")
+
+As a whole, I think the design is intuitive and not very verbose.  In particular,
+|| seems like a natural union operator because of its other CS definitions.  * Is
+a pretty common "zero or more" operator, and + has an expected behavior from other
+languages.  In addition, the literal extensions work well to save the user from
+concatenating many characters.
+
+On the otherhand, the design differs from other regex languages.  I'm not sure why
+we used the `<*>` operator, instead of just `*`, which is more common.  Same
+applies to `<+>`.  More problematic, though, was the concatenation operator. 
+Ideally, we would be able to concatenate expressions by simply writing them next
+to each other.  However, I'm not sure we would have been able to implement this
+behavior in Scala.  Also, it felt verbose using quotes to make a longer
+expression.  If I were actually implementing a regex DSL, I would probably have
+the user type a string, which would be parsed into a tree of Unions, Concats, and
```

Comment:
Because this is parsed, I think that makes it an external language. With the current assignment, it's interesting that that was your feeling when even creating the regex DSL. It seems like external DSLs often give us the exact behavior we want, but that they would definitely be more difficult to implement (for instance, our internal DSL does not require us to parse the input).